test/extensions/filters/network/rbac/integration_test.cc

```diff
@@ -129,6 +130,33 @@ name: rbac
   EXPECT_EQ(1U, test_server_->counter("tcp.rbac.denied")->value());
   EXPECT_EQ(1U, test_server_->counter("tcp.rbac.shadow_allowed")->value());
   EXPECT_EQ(0U, test_server_->counter("tcp.rbac.shadow_denied")->value());
+  // the detail is "none" because none of the policies is matched.
+  EXPECT_THAT(waitForAccessLog(listener_access_log_name_), testing::HasSubstr("RBAC none"));
+}
+
+TEST_P(RoleBasedAccessControlNetworkFilterIntegrationTest, DeniedWithDenyAction) {
+  useListenerAccessLog("%RESPONSE_FLAGS% %RESPONSE_CODE_DETAILS%");
+  initializeFilter(R"EOF(
+name: rbac
+typed_config:
+  "@type": type.googleapis.com/envoy.config.filter.network.rbac.v2.RBAC
+  stat_prefix: tcp.
+  rules:
+    action: DENY
+    policies:
+      "deny_all":
```

Comment:
Actually one other thought.
Are policy names allowed to have spaces and if so, do we think that would screw up people's access log parsing?  If so maybe we should substitute spaces for _s on the way out.  WDYT?

Reply:
yes the policy name could include spaces, actually it could be any character (e.g. `\n`, `\t`, etc.). 

I think we're going to break people anyway because the response_code_detail was changed from a fixed string `rbac_access_denied` to include the name of the policy, they need to update their parse code (if any) to adapt to the new format anyway. 

The space is probably not a big problem since people can change the logging format to use some delimiter for parse purpose (e.g. `|%RESPONSE_CODE_DETAILS%|`), also I think `RESPONSE_CODE_DETAILS` never claimed it will not include space?

Reply:
oh, I'm not worried about the string changing, I'm concerned that I think generally access logs are space delineated, so if any access logs have spaces and your parser says "the thing after the 4th space is the response code field" and the rc details has a space, there's a problem.

cc @dio as this falls under his purview of access logs.
Dio: is it expected any access logs additions are sanitized for spaces, or should we have a utility which does that on the way out?

Reply:
I think it is a good idea to preserve the number of "segments" in a log line. Yeah, I think we should have a way to do that.

Reply:
I added a new `replaceCharacters()` to the StringUtil and used it to replace all whitespaces with underscore, PTAL thanks,